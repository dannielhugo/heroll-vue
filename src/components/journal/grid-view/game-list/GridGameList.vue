<template>
  <div class="game-list">
    <div class="game-list__cols" v-for="game in games" :key="game.id">
      <GridGameItem :game="game" />
    </div>
  </div>
</template>

<script setup lang="ts">
import type { Game } from '@/models/game';
import GridGameItem from '../game-item/GridGameItem.vue';
defineProps<{ games: Game[] }>();
</script>

<style lang="scss" scoped>
.game-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 16px;
  grid-template-columns: repeat(auto-fit, 345px);
  justify-content: center;

  &__cols {
    margin-bottom: 20px;
    width: 345px;
    padding: 0;
  }
}
</style>
