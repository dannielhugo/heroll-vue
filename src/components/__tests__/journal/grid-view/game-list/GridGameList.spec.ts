import { mount } from '@vue/test-utils';
import ElementPlus from 'element-plus';
import GridGameList from '../../../../journal/grid-view/game-list/GridGameList.vue';
import * as game from '../../../mocks/Game.json';

describe('GridGameList', () => {
  test('should mount component', async () => {
    expect(GridGameList).toBeTruthy();

    const wrapper = mount(GridGameList, {
      props: {
        games: [game],
      },
      global: {
        plugins: [ElementPlus],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
