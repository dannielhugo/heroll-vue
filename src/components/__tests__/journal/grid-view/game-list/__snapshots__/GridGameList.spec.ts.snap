// Vitest Snapshot v1

exports[`GridGameList > should mount component 1`] = `
"<div class=\\"game-list\\" data-v-ef199a12=\\"\\">
  <div class=\\"game-list__cols\\" data-v-ef199a12=\\"\\">
    <div class=\\"el-card is-hover-shadow game-card\\" data-v-a56eedd2=\\"\\" data-v-ef199a12=\\"\\">
      <!--v-if-->
      <div class=\\"el-card__body\\" style=\\"padding: 0px;\\">
        <div class=\\"el-image\\" data-v-a56eedd2=\\"\\">
          <!--v-if-->
          <div class=\\"el-image__wrapper\\">
            <div class=\\"el-image__placeholder\\"></div>
          </div>
          <!--v-if-->
        </div>
        <div style=\\"padding: 14px;\\" data-v-a56eedd2=\\"\\">
          <div class=\\"game-card__title\\" data-v-a56eedd2=\\"\\">Soulcalibur (1998)</div>
          <div class=\\"game-card__subtitle\\" data-v-a56eedd2=\\"\\">Fighting</div>
          <div class=\\"game-card__subtitle\\" data-v-a56eedd2=\\"\\">Xbox 360, Dreamcast</div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
